app/controllers/rounds_controller.rb

```diff
@@ -19,6 +21,22 @@ def destroy
     destroy_resource
   end
 
+  api :POST, '/rounds/:round_id/allocations/upload', 'generates allocations for round from csv'
+  def upload
+    round = Round.find_by_id(params[:round_id])
+    if current_user.is_admin_for?(round.group)
+      csv = CSV.read(params[:csv].tempfile)
+      round.generate_allocations_from!(csv)
+      render status: 201, json: {
+        message: "upload successful, allocations created"
+      }
+    else 
+      render status: 403, json: {
+        error: "not an admin brah"
```

Comment:
lol. it'd be good if this error followed the same format as other [`policies`](https://github.com/cobudget/cobudget-api/tree/master/app/policies) (or if this could be a policy), even if cheeky.